### PR TITLE
DO NOT MERGE - Rails 5.2 + webpack

### PR DESCRIPTION
**Pull request by DmitryAvramec**
This PR includes Rails 5.2 upgrade which is not yet released. We can merge after release.
- [x]  fix javascript deps app/javascript/index.js
- [x]  fix css deps app/javascript/styles/app.css
- [x]  jquery-ui datepicker locales (can load all for now)
- [x] without scope on user
- [x] autocomlete on leads
- [x] Home#index
- [x]   token authenticity
- [x] $.effects()

### CODE REVIEW

**Comment by johnnyshields** [comment]
@CloCkWeRX just some background here. Rails 5.2+ is providing first class support to compiling JS/CSS assets via Webpack (Javascript lib). It is much faster than sprockets, especially in dev env.

File: Gemfile [hunk 1]
```diff
@@ -88,7 +89,7 @@ end
 gem 'rails', github: 'rails', branch: '5-2-stable'
 gem 'webpacker'
 gem 'foreman'
-gem 'rack-mini-profiler'
+gem 'rails-jquery-autocomplete'
```
Comment (johnnyshields) [inline Gemfile]: can we get rid of this gem?

**Comment by johnnyshields** [comment]
@DmitryAvramec the image files at app/javascript/app/images/jquery-ui/.../*.png should be included from the NPM repo of jquery UI

**Comment by johnnyshields** [comment]
This PR includes Rails 5.2 upgrade which is not officially released. We can merge when Rails 5.2 is released.

File: Gemfile [hunk 2]
```diff
@@ -85,6 +86,11 @@ group :heroku do
   gem 'rails_12factor'
 end
 
+gem 'rails', github: 'rails', branch: '5-2-stable'
+gem 'cancancan', github: 'lizzyaustad/cancancan', branch: 'rails_5_2_upgrade_fixes'
```
Comment (CloCkWeRX) [inline Gemfile]: https://github.com/CanCanCommunity/cancancan/blob/develop/CHANGELOG.md#220-apr-13th-2018 - can pin to < 3.0.0

File: Gemfile [hunk 3]
```diff
@@ -85,6 +86,11 @@ group :heroku do
   gem 'rails_12factor'
 end
 
+gem 'rails', github: 'rails', branch: '5-2-stable'
```
Comment (CloCkWeRX) [inline Gemfile]: Can swap to released versions

**Comment by CloCkWeRX** [comment]
I think with this one, I might just do the rails 5.2+ changes in isolation.

**Comment by johnnyshields** [comment]
We've abandoned this branch. If I remember correctly FFCRM was using quite a bit of CSS/JS templating features which work in Sprockets but not in Webpack. You are welcome to cherry-pick from this as needed.

**Comment by CloCkWeRX** [comment]
Going to closes off for now, landed rails 52 changes on master separately.

**Pull request closed without merging**